doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+        /// then install object semantics.
+        /// Likewise --Query from the command line is not implemented here since the query is required up 
+        /// front in order to get the CatalogPackage object.
+        /// DESIGN NOTE: Using PackageVersionInfo rather than CatalogPackage felt like it would be too 
+        /// specific here. CatalogPackage and PackageVersionId (or if unspecified then the latest one) may be nicer
+        /// if we end up having versions that have different applicability or other more complex logic about 
+        /// which version is preferred.
+        CatalogPackage CatalogPackage;
+        /// Optionally specifies the version from the package to install. If unspecified the version matching 
+        /// CatalogPackage.GetLatestVersion() is used.
+        PackageVersionId PackageVersionId;
+
+        /// Specifies alternate location to install package (if supported).
+        String PreferredInstallLocation;
+        /// User or Machine. 
+        AppInstallScope AppInstallScope;
+        /// Silent, Interactive, or Default
+        AppInstallMode AppInstallMode;
+        /// Directs the logging to a log file. Callers must provide a path to a file that the Windows Package 
+        /// Manager package has write access to.
+        String LogOutputPath;
+        /// Continues the install even if the hash in the catalog does not match the linked installer.
+        Boolean AllowHashMismatch;
+        /// A string that will be passed to the installer. 
+        /// IMPLEMENTATION NOTE: maps to "--override" in the winget cmd line
+        String ReplacementInstallerArguments;
+
+        /// Used by a caller to correlate the install with a caller's data.
+        /// IMPLEMENTATION NOTE: This area is in flux but right now it's expected that this would allow a caller to 
+        /// use JSON to encode multiple arguments.
+        String CorrelationData;
+        /// A string that will be passed to the source server if using a REST source
+        String AdditionalAppCatalogArguments;
+    }
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppInstaller
+    {
+        AppInstaller();
+
+        /// Get the available catalogs. Each source will have a separate catalog. This list does not include KnownCatalogs.
+        /// This does not open the catalog. These catalogs can be used individually or merged with GetCompositeAppCatalogAsync. 
+        /// IMPLEMENTATION NOTE: This is a list of sources returned by Windows Package Manager source list 
+        /// excluding the predefined sources.
+        Windows.Foundation.Collections.IVectorView<AppCatalog> GetAppCatalogs();
+        /// Get a built in catalog
+        AppCatalog GetAppCatalog(PredefinedAppCatalog predefinedAppCatalog);
+        /// Get a built in catalog
+        AppCatalog GetAppCatalogByLocalAppCatalog(LocalAppCatalog localAppCatalog);

```

Comment:
GetPredefinedAppCatalog and GetLocalAppCatalog.

Reply:
Yeah that's better